packages/@ama-sdk/client-angular/src/api-angular-client.ts

```diff
@@ -144,13 +149,14 @@ export class ApiAngularClient implements ApiClient {
 
         const subscription = httpRequest.subscribe({
           next: (res) => data = res,
+          // eslint-disable-next-line @typescript-eslint/prefer-promise-reject-errors -- subscription forwards the error from the httpRequest to asyncResponse promise
           error: (err) => reject(err),
           complete: () => resolve(data)
         });
         metadataSignal?.throwIfAborted();
         metadataSignal?.addEventListener('abort', () => {
           subscription.unsubscribe();
-          reject(metadataSignal.reason);
+          reject(metadataSignal.reason instanceof Error ? metadataSignal.reason : new Error(metadataSignal.reason.toString()));
```

Comment:
shouldn't the format look something like this?
```suggestion
          reject(metadataSignal.reason instanceof Error 
            ? metadataSignal.reason 
            : new Error(metadataSignal.reason.toString()));
```